### ISSUE TITLE
"Remove unnecessary parentheses" does not trigger for parenthesized simple constant value after binary OR operator

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#80777]
**Version Used**: 

**Steps to Reproduce**:
```csharp
class C
{
    public const int A = 1 | (2);
    public const int B = 1 | (2 | 3);
}
```

**Expected Behavior**:
The parentheses are greyed out in both expressions.

**Actual Behavior**:
The parentheses around the simple expression `(2)` are not detected as unnecessary (even though they are redundant and can safely be removed), while those around the more complex `(2 | 3)` expression are detected as unnecessary.

<img width="306" height="119" alt="Image" src="https://github.com/user-attachments/assets/fcd59fa2-8216-44d0-9e66-bd9246772324" />

The behaviour is correct for the + operator though:
<img width="312" height="114" alt="Image" src="https://github.com/user-attachments/assets/50a823e4-d387-45c8-8dd1-7c138449f74e" />
